eng/Publishing.props

```diff
@@ -26,4 +26,23 @@
   <ItemGroup>
     <Artifact Update="$(ArtifactsShippingPackagesDir)dotnet-sdk-pgo-*" Visibility="External" />
   </ItemGroup>
+
+  <Target Name="GetNonStableProductVersion">
+    <!-- Retrieve the non-stable product version. -->
+    <MSBuild Projects="$(RepoRoot)src/Installer/redist-installer/redist-installer.proj"
+             Targets="ReturnProductVersion">
+      <Output TaskParameter="TargetOutputs" PropertyName="NonStableProductVersion" />
+    </MSBuild>
+  </Target>
+
+  <Target Name="AddRelativeBlobPathToBlobArtifacts"
+          DependsOnTargets="GetNonStableProductVersion"
+          BeforeTargets="PublishToAzureDevOpsArtifacts"
+          AfterTargets="GenerateChecksumsFromArtifacts">
+    <ItemGroup>
+      <Artifact Condition="'%(Artifact.PublishFlatContainer)' == 'true' and '%(Artifact.RelativeBlobPath)' == ''"
```

Comment:
Can we update this check to check the Kind metadata as well?

Reply:
This is literally a copy-paste from windowsdesktop: https://github.com/dotnet/windowsdesktop/blob/02d6fe60db0c29ec1122f31190489f48ad40dd85/eng/Publishing.props#L60-L68

Should we / do we need to update aspnetcore, windowsdesktop and runtime with your suggestion?
What would I need to compare Kind against here?

Reply:
I think we need to compare `Kind == Blob`. Loose PDBs are already given a relative blob path with the build id, so those shouldn't need it.

Reply:
OK, updated. I like the Kind metadata.